### PR TITLE
Fix: DNSSEC - extract isZoneSecure and restore probeName stripping

### PR DESCRIPTION
**Pull request by MaciejTe**
Fix: DNSSEC - extract isZoneSecure and restore probeName stripping                                                                                                                    
                                                                
Deduplicate the "is zone signed?" check from answer() and authority() into a new isZoneSecure() method. The original refactoring (6143ad1) dropped the one-label-stripping logic before calling findDS, which could cause findDS to probe non-delegation-point host names and incorrectly conclude a signed zone is insecure. Restore the probeName stripping and add a debug log for the findDS error path to preserve diagnostic visibility.
                             
Add table-driven unit tests for isZoneSecure covering all branches (empty parentdsrr, DS-matches-zone fast path, case insensitivity, ancestor DS fallback, single-label qname, findDS error fail-closed) and integration tests for the answer/authority/NXDOMAIN paths.

### CODE REVIEW

File: middleware/resolver/resolver.go [hunk 1]
```diff
@@ -1277,6 +1251,51 @@ func (r *Resolver) findDS(ctx context.Context, signer, qname string, parentdsrr
...
+	// DS records only exist at delegation points (zone cuts), not at
+	// arbitrary owner names. Strip one label so that findDS probes
+	// delegation points rather than host names inside a zone.
+	probeName := qname
+	if labels := dns.SplitDomainName(qname); len(labels) > 1 {
+		probeName = dns.Fqdn(strings.Join(labels[1:], "."))
+	}
+
+	parentdsrr, err := r.findDS(ctx, "", probeName, parentdsrr)
+	if err != nil {
```
Comment (semihalev) [inline middleware/resolver/resolver.go:1299]: This changes the EDE code reported to clients. A network timeout during DS lookup would previously return its specific error (potentially NetworkError EDE), but now returns RRSIGsMissing EDE. This could make debugging harder. 

Consider either:
  - Returning (bool, error) from isZoneSecure() so the original error can be propagated
  - Or at least logging at Info level instead of Debug

**Comment by semihalev** [comment]
Can you rebase please.

**Comment by MaciejTe** [comment]
@semihalev I pushed DNSSEC validation improvement - `c-73-136-41-228.hsd1.tx.comcast.net.` (domain used in tests, test case it's used in started to fail) has DNSSEC configured, but has broken signing for some records (see https://dnssec-debugger.verisignlabs.com/comcast.net).

**Comment by MaciejTe** [comment]
@semihalev Any updates on this? I'm not sure about this solution - I'm fine if you close this MR, revert my previous PR and commit your own fix - the goal is to fix DNSSEC isssues on [dnssec.tools](https://dnscheck.tools/) website.

**Comment by semihalev** [comment]
I'm working on this.

**Comment by semihalev** [comment]
If you have time, can you check the flaky tests?

**Review by semihalev — APPROVED** [review]

**Comment by MaciejTe** [comment]
Thank you @semihalev !
